### PR TITLE
Anchor Link Fixes

### PR DESCRIPTION
**Pull request by jescalan**
I am excited to report that this PR takes the number of broken anchor links down from 611 to zero  🎉

The code changes here are mostly behind the scenes in changing the way that anchor links are generated and added to the document, there is only one new feature, which is **anchor link aliases**, which work like this:

```
# Headline ((#foo))
## Other Headline ((#bar, #baz))
```

Everything in between the double parens is removed from the output, and all the anchors are added as aliases for the headline. So that means going to `http://example.com/#headline` would work, and also `http://example.com/#foo` would bring the user to the same headline. This feature can be used to ease the pain of changing text contents over time. It is, of course, optional and should only be used when truly necessary.

I plan to quickly follow this PR with another one that will add a github check that will scan for broken links and anchor links any time that documentation files are changed. This should help to prevent accidental introduction of broken links in the future 💪🏼

### CODE REVIEW

File: website/package.json [hunk 1]
```diff
@@ -6,14 +6,14 @@
   "dependencies": {
     "@bugsnag/js": "^6.5.2",
     "@bugsnag/plugin-react": "^6.5.0",
-    "@hashicorp/nextjs-scripts": "^6.0.0-2",
+    "@hashicorp/nextjs-scripts": "^6.2.0-7",
```
Comment (kylemac) [inline website/package.json]: @jescalan are we able to get a non-prerelease on this PR?

Comment (jescalan) [reply website/package.json]: not quite yet unfortunately - we're still on prerelease for the css changes which this is based on top of. we can push those out as stable, but hashiconf and www are not on it yet. mike is working on getting them converted this week!

File: website/pages/docs/runtime/interpolation.mdx [hunk 2]
```diff
@@ -58,7 +58,7 @@ task "docs" {
 }
 ```
 
-## Node Variables [](#interpreted_node_vars)
+## Node Variables ((#interpreted_node_vars, #node-variables-))
```
Comment (jonathantneal) [inline website/pages/docs/runtime/interpolation.mdx]: I love the **anchor link aliases**. Thank you for documenting this in the PR.

**NIT**: I wasn’t clear whether one of these was the **primary** link. I now see that the primary link is generated from the heading.

**Comment by jescalan** [comment]
@pruett - it comes from our remark plugins, was built custom!

**Comment by DingoEatingFuzz** [comment]
Thank you for taking link preservation seriously, @jescalan! This looks good, but please forgive me for getting on my ol' soapbox again and saying this:

While this is syntactically very clean, it is also meaningless at first brush. What are those extra hash things? Why are they wrapped in double parens? Are the double parens necessary? 

Once an author has learned the language feature, I'm sure it's great and a pleasant authoring experience, but it represents another place where we are drifting from a well-documented standard. It's not hard to see a future code review where someone says "instead of adding an empty anchor, use this pattern instead". 

The only alternatives that would still be standards are existing HTML patterns (clunky, hard to read, easy to screw up) or perhaps an MDX component (also probably clunky, maybe clearer to read).

**tl;dr:** This gets a 👍from me, but any extension to markdown should be seen as an unfortunate compromise that increases the burden on content authors who now need to know markdown as well as our (hopefully well documented somewhere) collection of extensions.

**Comment by jescalan** [comment]
@DingoEatingFuzz Thank you for the review, and I totally agree here. The markdown extension is very much something that I wanted to avoid if at all possible. We just didn't really see a way that this could be done better which is why we settled with this. Just want to add some extra information here that will maybe help this to be a slightly easier pill to swallow.

We did consider both of the alternatives you mentioned above. With the MDX component option, the headline parsing and slug-ification happens at a different parse stage from where react components are handled, so in this case the only way to make this work like an MDX component would be to introduce syntax in as close to the same way as an MDX component as possible, then  manually parse it in a standard markdown plugin. This feels messy, difficult, and error prone, so we decided to pass on this.

The other option is the standard html option, which was a very close second. It would still need to be parsed and re-formatted into something different to align with our styling and a11y needs though, and parsing html in a markdown plugin as we all know is a bit of a dangerous situation. We'd also only be parsing for specific properties since we need to re-generate the html into the output, which could lead to authors being surprised that extra properties don't work. Also as you noted, html syntax is not very forgiving and across a large organization is prone to frequent syntax errors, which are almost always invisible since html elements don't render visibly, where errors to this custom syntax are very obvious, as the text will show up in the headline if you make a mistake.

The good news is that the current syntax is extremely find-and-replace friendly, as double parens are not a normal pattern anywhere else. The regex pattern for matching the syntax is extremely simple and robust, and even just a standard search for `((` across all of nomad identifies only these specific patterns. So, if we change our minds or need to back out of this down the line, I don't think the effort will be too heavy.

Additionally, we want to ensure that this syntax is using very sparingly, only in situations where a headline must be changed, and has a high volume of _external_ links coming in to its existing slug.  In this pull request, it was only added in places where there were existing manual html anchors already present, to avoid any unexpected regressions.

There are two things we plan to do to help this rare usage pattern to become a reality. First, making it clear in the readme where this syntax is documented that it is only to be used in this scenario, and that internal links to a changed anchor should be changed manually using find and replace if an anchor link changes. Second, we plan to implement a set of google analytics events that track anchor link hits and misses on any given page. This way, we will be able to check into how heavy of traffic any given anchor link is getting, and where its coming from so that we can make these decisions in a more informed manner.

Hope this is helpful detail, and I will get on adding those extra readme details to this pull request now! I had them elsewhere but it makes way more sense to include them here, so thank you for the reminder 😀

**Comment by DingoEatingFuzz** [comment]
Thank you for the insights into the deliberation process Jeff! Sounds like you have all taken this very seriously, which is always great to hear. As always, I appreciate the discourse!
